### PR TITLE
oci conmon: tell conmon to log container name

### PR DESCRIPTION
**Pull request by haircommander**
specifying `-n=ctr-name` tells conmon to log CONTAINER_NAME=name if the log driver is journald

add this, and a test!

also, refactor the args slice creation to not append() unnecessarily.

technically, this needs https://github.com/containers/conmon/pull/154, though the test is skipped anyway

Signed-off-by: Peter Hunt <pehunt@redhat.com>

### CODE REVIEW

File: test/e2e/logs_test.go [hunk 1]
```diff
@@ -173,6 +173,24 @@ var _ = Describe("Podman logs", func() {
 		Expect(string(out)).To(ContainSubstring("alpine"))
 	})
 
+	It("podman journald logs for container name", func() {
+		Skip("need to verify images have correct packages for journald")
```
Comment (rhatdan) [inline test/e2e/logs_test.go:177]: You test is always skipped?

**Comment by rhatdan** [comment]
LGTM
other then the issue with skipping the test.

File: test/e2e/logs_test.go (hunk 1 above)
Comment (haircommander) [reply test/e2e/logs_test.go:177]: all of the logs tests are always skipped

Comment (haircommander) [reply test/e2e/logs_test.go:177]: they're actually skipped in two different spots :smile:

**Comment by haircommander** [comment]
all logs tests are skipped [here](https://github.com/containers/libpod/blob/a6700147a820da5662ebe114a89c644df4fe1845/test/e2e/logs_test.go#L22)
all journald tests are skipped [here](https://github.com/containers/libpod/blob/a6700147a820da5662ebe114a89c644df4fe1845/test/e2e/logs_test.go#L160)

I have a feeling fully enabling the test will be a bit more challenging than dropping these. do you two want me to address those here @rhatdan @vrothberg ?

**Comment by vrothberg** [comment]
> all logs tests are skipped here

Only for remote though.

> I have a feeling fully enabling the test will be a bit more challenging than dropping these. do you two want me to address those here @rhatdan @vrothberg ?

If you have some cycles for that, it would be great. Adding disabled tests is not ideal.

**Comment by haircommander** [comment]
I have about enough cycles to attempt to enable them, but I can't promise carrying through fixing the tests if they fail from v2/not having journalctl on the images

**Comment by haircommander** [comment]
(also without https://github.com/containers/conmon/pull/154 the test will fail)

**Comment by haircommander** [comment]
update: ci images do not have journald. I would like to enable the tests as a follow up

**Pull request merged**